File: Frontend/src/app/model/Tables.ts
import { Restaurant } from "./Restaurant";

export class Tables {
  idTable!: number;
  numTable!: number;
  nbplaceT!: number;
  etatT!: string;
  cuisine!: string;
  reservation!: Date;
  idRestaurant!: number;
  restaurant?: Restaurant | undefined; // Include undefined in the type

  // Additional method to set restaurant ID
}

File: MicroService_back-RayenJouida/GestionRestaurant/src/main/java/twin/microservice/gestionrestaurant/entities/Tables.java
package twin.microservice.gestionrestaurant.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.util.Date;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Tables implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idTable;
    private Long numTable;
    private Integer nbplaceT;
    private String etatT;
    private String cuisine;
    private Date reservation;


    @JsonIgnore

    @ManyToOne
    @JoinColumn(name = "restaurant_id")
    private Restaurant restaurant;
    public void setRestaurantId(Long restaurantId) {
        if (this.restaurant == null) {
            this.restaurant = new Restaurant();
        }
        this.restaurant.setIdRestaurant(restaurantId);
    }

}
